### PR TITLE
Add parsing for mark and unmark command

### LINKED ISSUE #135: Add parsing for mark and unmark

### PR DESCRIPTION
**Pull request by jxt00**
Closes #135 

* All instances of `isAllStudents` have been removed.
* Change `Optional<List<Student>>` to `List<Student>`.

To be merged only after #139 is merged.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        try {
+            Index classGroupIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_CLASS_INDEX).get());
+            Index weekIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_WEEK).get());
+            ObservableList<Student> students = ParserUtil.parseStudents(
+                    argMultimap.getValue(PREFIX_STUDENT).get(), model);
+            boolean isAllStudents = false;
+            if (students.size() == model.getFilteredStudentList().size()) {
+                isAllStudents = true;
+            }
```
Comment (lkldev) [inline src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java:59]: yeah from the way it seems, isAllStudents seems redundant, so I guess we can remove isAllStudents then, and remove the Optional.of 

P.S. will need to edit for Mark, Unmark, Grade and Enrol, Disenrol command

**Review by lkldev — COMMENTED** [review]
Some comments on the classes, for the filtering we can discuss tmr

File: src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java (hunk 1 above)
Comment (jxt00) [reply src/main/java/seedu/address/logic/parser/UnmarkCommandParser.java]: Yep `isAllStudents` has been removed from the aforementioned classes.

**Review by lkldev — APPROVED** [review]
LGTM :)

**Review by wxliong — APPROVED** [review]
LGTM! :)

**Pull request merged**